shared/gh/js/views/gh.series-info.js

```diff
@@ -0,0 +1,221 @@
+/*!
+ * Copyright 2015 Digital Services, University of Cambridge Licensed
+ * under the Educational Community License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['gh.core', 'moment', 'clickover'], function(gh, moment) {
+
+    // Cache the retrieved series
+    var series = {};
+
+    /**
+     * Retrieve the series information
+     *
+     * @param  {Number}    seriesId    The ID of the series that needs to be retrieved
+     * @private
+     */
+    var retrieveSeries = function(seriesId) {
+
+        // Fetch the series if it hasn't been retrieved yet
+        if (!series[seriesId]) {
+            return gh.api.seriesAPI.getSeries(seriesId, false, function(err, data) {
+                if (err) {
+                    return console.error(err);
+                }
+                series[seriesId] = data;
+
+                // Retrieve the events that belong to the series
+                if (!series[seriesId]['Events']) {
+                    series[seriesId]['Events'] = [];
```

Comment:
Instead of continuing the christmas tree could we get away with doing:
```javascript
// Create an empty array if the series has no events
series[seriesId]['Events'] = series[seriesId]['Events'] ? series[seriesId]['Events'] : []
```